File: src/endpoint-tests/Bc.Endpoint.Integration.Tests/EndpointFactory.cs
using System.Threading.Tasks;
using Bc.Common.Endpoint;
using Bc.Contracts.Internals.Endpoint.CommentTaking.Commands;
using Bc.Contracts.Internals.Endpoint.GitHubPullRequestCreation.Messages;
using NServiceBus;

namespace Bc.Endpoint.Integration.Tests
{
    internal static class EndpointFactory
    {
        internal static Task<IEndpointInstance> GetSenderEndpoint()
        {
            const string bcEndpointAssemblyName = "Bc.Endpoint";

            var endpoint = EndpointCommon.GetEndpoint(
                "Sender.Tests",
                false,
                new EndpointCommonConfigurationProvider());

            var scanner = endpoint.AssemblyScanner();
            scanner.ExcludeAssemblies(bcEndpointAssemblyName);

            var transport = endpoint.UseTransport<SqlServerTransport>();
            var routing = transport.Routing();
            routing.RouteToEndpoint(typeof(RequestCreateGitHubPullRequest).Assembly, bcEndpointAssemblyName);
            routing.RouteToEndpoint(typeof(TakeComment).Assembly, bcEndpointAssemblyName);

            return NServiceBus.Endpoint.Start(endpoint);
        }
    }
}

File: src/tests/integration/Bc.Endpoint.Integration.Tests/EndpointFactory.fs
module Bc.Endpoint.Integration.Tests.EndpointFactory

open System.Threading.Tasks
open Bc.Common.Endpoint
open Bc.Contracts.Internals.Endpoint.CommentTaking.Commands
open Bc.Contracts.Internals.Endpoint.GitHubPullRequestCreation.Messages
open NServiceBus

let getSenderEndpoint () =
    let bcEndpointAssemblyName = "Bc.Endpoint"

    let endpoint = getEndpoint "Sender.Tests" false

    let scanner = endpoint.AssemblyScanner()
    scanner.ExcludeAssemblies(bcEndpointAssemblyName)

    // routing
    let setRouting (transport: TransportExtensions<'T>) =
       transport.Routing().RouteToEndpoint((typeof<TakeComment>).Assembly, bcEndpointAssemblyName)

    match ConfigurationProvider.isUseLearningTransportAndPersistence with
    | false -> setRouting (endpoint.UseTransport<SqlServerTransport>())
    | true -> setRouting (endpoint.UseTransport<LearningTransport>())

    NServiceBus.Endpoint.Start(endpoint)